packages/staking/solana/lib/rewards.ts

```diff
@@ -207,3 +224,16 @@ export const calcRewards = (
 
   return rewardEntryAccumulator.getClaimableAmount();
 };
+
+export const calculateRewardAmount = (rewardRate: number, stakeTokenDecimals: number, rewardTokenDecimals: number) => {
+  const rewardTokenValue = getBN(rewardRate, rewardTokenDecimals);
+  const decimalsDiff = REWARD_AMOUNT_DECIMALS - stakeTokenDecimals;
+  if (decimalsDiff === 0) {
+    return rewardTokenValue;
+  }
+  const diffFactor = new BN(10).pow(new BN(Math.abs(decimalsDiff)));
+  if (decimalsDiff > 0) {
+    return rewardTokenValue.mul(diffFactor);
+  }
+  return rewardTokenValue.div(diffFactor);
+};
```

Comment:
I'd add another one with only 2 arguments: (rewardRateRaw: BN, stakeTokenDecimals: number) => because it is basically what you have in the end of the day. 

What about backwards transformation? Do you think it is worth exposing it from js-sdk? 
like from on-chain values to the actual raw reward rate (in reward tokens)